src/modules/dashboard/components/Dialogs/CreatePaymentDialog/CreatePaymentDialog.tsx

```diff
@@ -93,6 +98,42 @@ const CreatePaymentDialog = ({
     variables: { colonyAddress },
   });
 
+  const { data: colonyData } = useColonyFromNameQuery({
+    variables: { name: colonyName, address: colonyAddress },
+  });
+  const isWhiteListActive = colonyData?.processedColony?.isWhitelistActivated;
```

Comment:
You already have this colony object passed as a prop to this modal, why are you calling the query for it again ?

This is super expensive. _(in terms of requests made to the server)_

Reply:
Well spotted. You're correct. I removed the extra calls using `useColonyFromNameQuery` & `useVerifiedUsersQuery`.